variod.c

```diff
@@ -255,6 +255,20 @@ int main(int argc, char *argv[])
 	
 	// setup and start pcm player
 	start_pcm();
+	// Open Socket for TCP/IP communication to XCSoar
+	xcsoar_sock = socket(AF_INET, SOCK_STREAM, 0);
+	if (xcsoar_sock == -1)
+		fprintf(stderr, "could not create socket\n");
+	  
+	s_xcsoar.sin_addr.s_addr = inet_addr("127.0.0.1");
+	s_xcsoar.sin_family = AF_INET;
+	s_xcsoar.sin_port = htons(4352);
+
+	// try to connect to XCSoar
+	wait_for_XCSoar(xcsoar_sock, (struct sockaddr*)&s_xcsoar);
+	// make socket to XCsoar non-blocking
+	fcntl(xcsoar_sock, F_SETFL, O_NONBLOCK);
```

Comment:
Let's move code to connect to xcsoar to separate function that returns a socket to avoid duplication:

```c
int connect_to_xcsoar() 
{
...
}

Reply:
Perfect. I'll move it there.